### PR TITLE
chore: build FE container at build time

### PR DESCRIPTION
**Pull request by ebezzi**
#3830

## Changes

- Builds the FE container at build time and not at runtime (like currently)
- Lower the memory requirement to 4G (minimum for the FARGATE configuration we have)
- Revert the healthcheck interval to 15s

## Testing steps
- https://ebezzi-fe-build-frontend.rdev.single-cell.czi.technology/
- Test the local application (it should work with `make local-sync`)
- Check that it works in dev and the container boots in a few seconds (through Cloudwatch)
- Check that the sitemaps are working (they're affected by this change)

### CODE REVIEW

**Comment by seve** [comment]
Can we leave the `configs/*.js` files intact for development purposes? I'd prefer to not have to manually set up my `config.js` when running the dev server

File: frontend/Dockerfile [hunk 1]
```diff
@@ -15,22 +15,21 @@ RUN apt-get update && apt-get install -y make wget \
...
 ENV PATH /opt/node_app/node_modules/.bin:$PATH
 
 # -- TODO, we should try turning this back on later.
 # ADD --chown=node . /corpora-frontend
 ADD . /corpora-frontend
 WORKDIR /corpora-frontend
+ADD ./src/configs/build.js src/configs/configs.js
+RUN mkdir -p node_modules
+RUN ln -sf /opt/node_app/node_modules/* /opt/node_app/node_modules/.bin ./node_modules/.
+RUN npm run build
```
Comment (ebezzi) [inline frontend/Dockerfile:32]: @jgadling we had to replace `next build` with this so that the `postbuild` step could run. Any concerns?

Comment (jgadling) [reply frontend/Dockerfile:32]: Nope, I just didn't know what the best way to build the static app was!

File: frontend/src/configs/local.js [hunk 2]
```diff
@@ -1,11 +1,16 @@
-const { PLAUSIBLE_DATA_DOMAIN_STAGING } = require("./common");
-
 // (thuang): For local development, please copy the content of this file
 // to a new file named `configs.js` in this directory.
 
 const configs = {
+  // Dev
```
Comment (tihuan) [inline frontend/src/configs/local.js:5]: @seve thanks for pointing this out!

I just added different endpoints in `local.js`, so we can still copy this to `configs.js` locally and run `npm run dev` without using `make local-sync`. So this should address our `npm run dev` use case now?

Thanks!

Comment (seve) [reply frontend/src/configs/local.js:5]: Very much a nit, but the process of:
```
cp src/configs/dev.js src/configs/configs.js
npm run dev
```
is **way** faster than
```
cp src/configs/local.js src/configs/configs.js
code src/configs/configs.js
# uncomment one line and comment another line
npm run dev
```

Comment (tihuan) [reply frontend/src/configs/local.js:5]: yeah I actually just changed the default in `local.js` to be pointing to the dev API

**Comment by tihuan** [comment]
Also a huge shout out to @jgadling again for coming up with the solution AND the code to make this happen 🙏 ❤️ We're able to decrease Fargate memory from 8GB to 4GB because of this change too, so it's literally saving CZI money 💰 

And of course @ebezzi for pairing with me today to ironed out the sitemap gen issue I didn't catch earlier + changing our Infra config 🙏

Very grateful to be working with you both!

File: frontend/src/configs/local.js (hunk 2 above)
Comment (seve) [reply frontend/src/configs/local.js:5]: oh missed that part, sounds good!

**Review by seve — APPROVED** [review]

**Review by tihuan — APPROVED** [review]
LGTMMM 🎶 🎶 🎶

**Comment by tihuan** [comment]
`backend-wmg-unit-test` and `wmg-processing-unit-test` failures are being taken care of and irrelevant to this PR. So I'll merge it now!

**Pull request merged**